### ISSUE TITLE
Unique validation fails due to matching itself on update

### ISSUE THREAD

**Issue by darends** [unite-cms/unite-cms#502]
The functions written for #457 have provided a very nice validation method to check the uniqueness of a field within a content type.

It works perfectly when creating an entry.
But apon editing the entry and attempting to re-save, it fails.

This is due to `content_unique` matching on itself

```json
{
    "expression": "content_unique(content.data.slug,\"slug\")",
    "message": "This Slug is not unique. Please chose a different one",
    "path": "slug",
    "groups": [
        "CREATE",
        "UPDATE"
    ]
}
```

Having unique validation on update is just as important as on create. 

-Dirk

**Comment by franzwilding** [comment]
You are right! Thank you for the report. Do you want to try to create a PR for this?

**Comment by darends** [comment]
A question i have regarding this issue.

I suggest to extend the query in `/unite-cms/src/Bundle/CoreBundle/Expression/UniteExpressionLanguageDoctrineContentProvider.php` like so
```php
    private function contentDataUnique($value, $path) : bool {
        $query = $this->entityManager->createQueryBuilder()
            ->select('count(c.id)')
            ->where('c.contentType = :contentType')
            ->andWhere('JSON_EXTRACT(c.data, :identifier) = :value')
            ->andWhere('c.id != :id')                              # Add line
            ->from('UniteCMSCoreBundle:Content', 'c')
            ->setParameters(
                [
                    ':contentType' => $this->contentType,
                    ':identifier' => '$.'.$path,
                    ':value' => $value,
                    ':id' => $id,                                  # Add line
                ]
            )
            ->getQuery();
```

The question i have is, what is the best way to pass the entry's ID to this function.

* Is it available through Doctrine as in identifier
* or should it be passed from the validation definition
resulting in `private function contentDataUnique($value, $path, $id) : bool {`

 Should this also be applied to `contentDataUniquify` also?
-Dirk

**Comment by darends** [comment]
Just wondering if you could provide any direction on the best way to get the content entry's ID.

**Comment by franzwilding** [comment]
In PHP, you can just use the getter:

$content = new Content();
$content->getId();

Is this what you are looking for?

**Comment by franzwilding** [comment]
I think you could also pass the whole content object to the function and access the id like in the prev post

**Comment by franzwilding** [comment]
Hey @darends 

Once https://github.com/unite-cms/unite-cms/pull/536 is merged into master, you can do the following: 
```
{
          "expression": "content_unique(content.data.slug.text,\"slug.text\", [content.id])",
          "message": "This Slug is not unique. Please chose a different one",
          "path": "slug",
          "groups": [
            "CREATE",
            "UPDATE"
          ]
        }
```
